Indicators/StandardDeviation.cs

```diff
@@ -0,0 +1,72 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); 
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using System.Linq;
+using System.Collections.Generic;
+using MathNetStatistics = MathNet.Numerics.Statistics.Statistics;
+
+namespace QuantConnect.Indicators
+{
+    public class StandardDeviation : WindowIndicator<IndicatorDataPoint>
+    {
+        /// <summary>
+        /// Initializes a new instance of the StandardDeviation class with the specified period.
+        /// 
+        /// Evaluates the standard deviation of samples in the lookback period. 
+        /// On a dataset of size N will use an N normalizer and would thus be biased if applied to a subset.
+        /// </summary>
+        /// <param name="period">The sample size of the standard deviation</param>
+        public StandardDeviation(int period)
+            : this("STD" + period, period)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the StandardDeviation class with the specified period.
+        /// 
+        /// Evaluates the standard deviation of samples in the lookback period. 
+        /// On a dataset of size N will use an N normalizer and would thus be biased if applied to a subset.
+        /// </summary>
+        /// <param name="name">The name of this indicator</param>
+        /// <param name="period">The sample size of the standard deviation</param>
+        public StandardDeviation(string name, int period)
+            : base(name, period)
+        {
+        }
+
+        /// <summary>
+        /// Gets a flag indicating when this indicator is ready and fully initialized
+        /// </summary>
+        public override bool IsReady {
+            get { return Samples >= Period; }
+        }
+
+        /// <summary>
+        /// Computes the next value of this indicator from the given state
+        /// </summary>
+        /// <param name="input">The input given to the indicator</param>
+        /// <returns>A new value for this indicator</returns>
+        protected override decimal ComputeNextValue(IReadOnlyWindow<IndicatorDataPoint> window, IndicatorDataPoint input)
+        {
+            if (!IsReady) {
+                return input;
```

Comment:
I think we want to return 0 for the zeroth sample and the first sample,

```csharp
if (Samples < 2)
{
    return 0m;
}
```

for any other number of samples, we can use compute the standard deviation in the same manner as below. The `IsReady` flag will let consumers know when we have enough samples for the n period std deviation.